provider/server/exec.go

```diff
@@ -149,7 +153,7 @@ func (e executorExec) Exec(ctx context.Context, in io.Reader, out io.WriteCloser
 			n, err := inputReader.Read(buf)
 			if err != nil {
 				if _, ok := err.(term.EscapeError); ok {
-					_, _ = f.Write(buf) // This is to close the writer.
+					f.Close() // This is to close the writer.
```

Comment:
why is it here? i expect it on end of function  behind wg.Wait

Reply:
or ideally use defer

Reply:
The current position of f.Close() is appropriate, as it ensures that the file descriptor is closed at the right moment, when the escape sequence is received. Moving f.Close() to a defer statement within the goroutine would not work as intended. Deferring f.Close() would mean that it is called when the goroutine exits, which is not desirable in this case. The f.Close() call needs to be executed immediately when the escape sequence is detected to ensure that the writer (the goroutine reading from in) exits the infinite loop and the file descriptor is closed promptly.